File: packages/ts/src/modules/marketplace.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse } from '../types.js';

export class MarketplaceModule {
  constructor(private readonly client: HubflyClient) {}

  public async getTemplates(): Promise<ApiResponse<unknown[]>> {
    return this.client.request<unknown[]>('/api/v1/marketplace/templates');
  }
}

File: packages/go/marketplace.go
package hubfly

import "context"

type MarketplaceService struct {
	client *Client
}

func (s *MarketplaceService) GetTemplates(ctx context.Context) (*Response[[]interface{}], error) {
	return Request[[]interface{}](ctx, s.client, "GET", "/api/v1/marketplace/templates", nil)
}
